datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVUtil.scala

```diff
@@ -0,0 +1,615 @@
+      val parentColumnId = parentColumn.getColumnId
+      columnTableRelation = ColumnTableRelation(parentColumnName = parentColumnName,
+        parentColumnId = parentColumnId,
+        parentTableName = parentTableName,
+        parentDatabaseName = parentDatabaseName, parentTableId = parentTableId)
+      columnTableRelation
+    } else {
+      columnTableRelation
+    }
+  }
+
+  private def getFields(dataType: DataType,
+      aggregateType: String,
+      columnTableRelationList: Seq[ColumnTableRelation],
+      actualColumnName: String) = {
+    val rawSchema = '`' + actualColumnName + '`' + ' ' + dataType.typeName
+    val dataMapField = DataMapField(aggregateType, Some(columnTableRelationList))
+    if (dataType.typeName.startsWith("decimal")) {
+      val (precision, scale) = CommonUtil.getScaleAndPrecision(dataType.catalogString)
+      (Field(column = actualColumnName,
+        dataType = Some(dataType.typeName),
+        name = Some(actualColumnName),
+        children = None,
+        precision = precision,
+        scale = scale,
+        rawSchema = rawSchema), dataMapField)
+    } else {
+      (Field(column = actualColumnName,
+        dataType = Some(dataType.typeName),
+        name = Some(actualColumnName),
+        children = None,
+        rawSchema = rawSchema), dataMapField)
+    }
+  }
+
+  private def getFieldToDataMapFields(name: String, dataType: DataType,
+      aggregateType: String,
+      arrayBuffer: ArrayBuffer[ColumnTableRelation], parenTableName: String) = {
+    var actualColumnName = name.replace("(", "_")
+      .replace(")", "")
+      .replace(" ", "_")
+      .replace("=", "")
+      .replace(",", "")
+    if (aggregateType.isEmpty && !parenTableName.isEmpty) {
+      actualColumnName = parenTableName + "_" + actualColumnName
+    }
+    getFields(dataType, aggregateType, arrayBuffer, actualColumnName)
+  }
+
+  private def getFieldsFromAggregate(carbonTable: CarbonTable,
+      groupByExp: Seq[Expression],
+      aggExp: Seq[NamedExpression],
+      fieldToDataMapFieldMap: mutable.LinkedHashMap[Field, DataMapField]): scala.collection
+  .mutable.LinkedHashMap[Field, DataMapField] = {
+    aggExp.map { agg =>
+      var aggregateType: String = ""
+      val arrayBuffer: ArrayBuffer[ColumnTableRelation] = new ArrayBuffer[ColumnTableRelation]()
+      var tableName = carbonTable.getTableName
+      var columnName = agg.name
+      agg.collect {
+        case attr: AttributeReference =>
+          arrayBuffer += getColumnRelation(attr.name,
+            carbonTable.getAbsoluteTableIdentifier.getCarbonTableIdentifier.getTableId,
+            carbonTable.getAbsoluteTableIdentifier.getCarbonTableIdentifier.getTableName,
+            carbonTable.getAbsoluteTableIdentifier.getCarbonTableIdentifier.getDatabaseName,
+            carbonTable)
+        case Alias(attr: AggregateExpression, name) =>
+          aggregateType = attr.aggregateFunction.nodeName
+        case Alias(_, name) =>
```

Comment:
I am not sure of the use of the above 2 cases.  why not a single case is enough?